src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/DiagnosticListener.cs

```diff
@@ -196,7 +213,9 @@ public override void Write(string name, object value)
         private class DiagnosticSubscription : IDisposable
         {
             internal IObserver<KeyValuePair<string, object>> Observer;
+
             internal Predicate<string> IsEnabled;
+            internal Func<string, object, object, bool> IsEnabledExt;
```

Comment:
Please rename IsEnabled -> IsEnabled1Arg, and IsEnabled3Arg  Also place a comment that any subscriber must set BOTH of these delegates (either they are both null or both are valid (typically one delegates to the other).    ALso indicate the rationale (we do this so that if both the source and receiver agree on the number of args, dispatch is very efficient.   Only in a mismatch is a 'fixup' thunk called.